plugins/nvim-lsp/basic-servers.nix

```diff
@@ -225,6 +226,58 @@ with lib; let
       description = "Enable lua LS, for lua";
       package = pkgs.lua-language-server;
       serverName = "lua_ls";
+
+      # All available settings are documented here:
+      # https://github.com/LuaLS/lua-language-server/wiki/Settings
+      settingsOptions = {
+        runtime = {
+          version = mkOption {
+            type = types.str;
+            description = ''
+              Tell the language server which version of Lua you're using
+              (most likely LuaJIT in the case of Neovim)
+            '';
+            default = "LuaJIT";
```

Comment:
I feel like still having types.nullOr is nice because then a user has a way to set the  value to the 'default' according to the LSP without searching for the default.